From 6ec197846a7d24c40094d1d4d4fa1b8cb09a485b Mon Sep 17 00:00:00 2001
From: Lukasz Kazmierczak <lukasz.kazmierczak@intel.com>
Date: Thu, 23 Mar 2023 16:52:56 +0100
Subject: [PATCH 1/2] cups-service schema XML files added

OEM schema implemented for cups-service, proper schema XMLs are added,
so RedfishServiceValidator can verify CUPS nodes for conformance.

Signed-off-by: Alicja Rybak <alicja.rybak@intel.com>
Signed-off-by: Lukasz Kazmierczak <lukasz.kazmierczak@intel.com>
Upstream-Status: Pending
---
 static/redfish/v1/$metadata/index.xml         |  18 +++
 .../v1/schema/CupsPolicyCollection_v1.xml     |  66 ++++++++
 static/redfish/v1/schema/CupsPolicy_v1.xml    | 109 +++++++++++++
 .../v1/schema/CupsSensorCollection_v1.xml     |  65 ++++++++
 static/redfish/v1/schema/CupsService_v1.xml   | 143 ++++++++++++++++++
 .../redfish/v1/schema/OemServiceRoot_v1.xml   |  53 +++++++
 6 files changed, 454 insertions(+)
 create mode 100644 static/redfish/v1/schema/CupsPolicyCollection_v1.xml
 create mode 100644 static/redfish/v1/schema/CupsPolicy_v1.xml
 create mode 100644 static/redfish/v1/schema/CupsSensorCollection_v1.xml
 create mode 100644 static/redfish/v1/schema/CupsService_v1.xml
 create mode 100644 static/redfish/v1/schema/OemServiceRoot_v1.xml

diff --git a/static/redfish/v1/$metadata/index.xml b/static/redfish/v1/$metadata/index.xml
index c9acf345..42ccb9e6 100644
--- a/static/redfish/v1/$metadata/index.xml
+++ b/static/redfish/v1/$metadata/index.xml
@@ -3198,4 +3198,22 @@
         <edmx:Include Namespace="OemLicense"/>
         <edmx:Include Namespace="OemLicense.v1_0_0"/>
     </edmx:Reference>
+    <edmx:Reference Uri="/redfish/v1/schema/OemServiceRoot_v1.xml">
+        <edmx:Include Namespace="OemServiceRoot"/>
+        <edmx:Include Namespace="OemServiceRoot.v1_0_0"/>
+    </edmx:Reference>
+    <edmx:Reference Uri="/redfish/v1/schema/CupsService_v1.xml">
+        <edmx:Include Namespace="CupsService"/>
+        <edmx:Include Namespace="CupsService.v1_0_0"/>
+    </edmx:Reference>
+    <edmx:Reference Uri="/redfish/v1/schema/CupsSensorCollection_v1.xml">
+        <edmx:Include Namespace="CupsSensorCollection"/>
+    </edmx:Reference>
+    <edmx:Reference Uri="/redfish/v1/schema/CupsPolicy_v1.xml">
+        <edmx:Include Namespace="CupsPolicy"/>
+        <edmx:Include Namespace="CupsPolicy.v1_0_0"/>
+    </edmx:Reference>
+    <edmx:Reference Uri="/redfish/v1/schema/CupsPolicyCollection_v1.xml">
+        <edmx:Include Namespace="CupsPolicyCollection"/>
+    </edmx:Reference>
 </edmx:Edmx>
diff --git a/static/redfish/v1/schema/CupsPolicyCollection_v1.xml b/static/redfish/v1/schema/CupsPolicyCollection_v1.xml
new file mode 100644
index 00000000..d5e17a37
--- /dev/null
+++ b/static/redfish/v1/schema/CupsPolicyCollection_v1.xml
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!---->
+<!--################################################################################       -->
+<!--# Redfish Schema:  CupsPolicyCollection                                                -->
+<!--#                                                                                      -->
+<!--################################################################################       -->
+<!---->
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="OData"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Capabilities.V1.xml">
+    <edmx:Include Namespace="Org.OData.Capabilities.V1" Alias="Capabilities"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Resource_v1.xml">
+    <edmx:Include Namespace="Resource.v1_0_0"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/RedfishExtensions_v1.xml">
+    <edmx:Include Namespace="RedfishExtensions.v1_0_0" Alias="Redfish"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="/redfish/v1/schema/CupsPolicy.xml">
+    <edmx:Include Namespace="CupsPolicy"/>
+    <edmx:Include Namespace="CupsPolicy.v1_0_0"/>
+  </edmx:Reference>
+
+  <edmx:DataServices>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsPolicyCollection">
+      <Annotation Term="Redfish.OwningEntity" String="DMTF"/>
+
+      <EntityType Name="CupsPolicyCollection" BaseType="Resource.v1_0_0.ResourceCollection">
+        <Annotation Term="OData.Description" String="A Collection of CUPS Policy Resource instances."/>
+        <Annotation Term="OData.LongDescription" String="This Resource shall represent a Resource Collection of CUPS Policy instances for a Redfish implementation."/>
+        <Annotation Term="Capabilities.InsertRestrictions">
+          <Record>
+            <PropertyValue Property="Insertable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.UpdateRestrictions">
+          <Record>
+            <PropertyValue Property="Updatable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.DeleteRestrictions">
+          <Record>
+            <PropertyValue Property="Deletable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Redfish.Uris">
+          <Collection>
+            <String>/redfish/v1/Oem/Intel/CupsService/Policies</String>
+          </Collection>
+        </Annotation>
+        <NavigationProperty Name="Members" Type="Collection(CupsPolicy.CupsPolicy)">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/Read"/>
+          <Annotation Term="OData.Description" String="The members of this collection."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain an array of links to the members of this collection."/>
+          <Annotation Term="OData.AutoExpandReferences"/>
+          <Annotation Term="Redfish.Required"/>
+        </NavigationProperty>
+      </EntityType>
+
+    </Schema>
+  </edmx:DataServices>
+</edmx:Edmx>
diff --git a/static/redfish/v1/schema/CupsPolicy_v1.xml b/static/redfish/v1/schema/CupsPolicy_v1.xml
new file mode 100644
index 00000000..be453a59
--- /dev/null
+++ b/static/redfish/v1/schema/CupsPolicy_v1.xml
@@ -0,0 +1,109 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!---->
+<!--################################################################################       -->
+<!--# Redfish Schema:  CupsPolicy                                                          -->
+<!--#                                                                                      -->
+<!--################################################################################       -->
+<!---->
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="OData"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Capabilities.V1.xml">
+    <edmx:Include Namespace="Org.OData.Capabilities.V1" Alias="Capabilities"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Measures.V1.xml">
+    <edmx:Include Namespace="Org.OData.Measures.V1" Alias="Measures"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/RedfishExtensions_v1.xml">
+    <edmx:Include Namespace="RedfishExtensions.v1_0_0" Alias="Redfish"/>
+    <edmx:Include Namespace="Validation.v1_0_0" Alias="Validation"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Resource_v1.xml">
+    <edmx:Include Namespace="Resource"/>
+    <edmx:Include Namespace="Resource.v1_0_0"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="/redfish/v1/schema/CupsSensor_v1.xml">
+    <edmx:Include Namespace="CupsSensor"/>
+    <edmx:Include Namespace="CupsSensor.v1_0_0"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="/redfish/v1/schema/CupsPolicy_v1.xml">
+    <edmx:Include Namespace="CupsPolicy"/>
+    <edmx:Include Namespace="CupsPolicy.v1_0_0"/>
+  </edmx:Reference>
+
+  <edmx:DataServices>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsPolicy">
+      <Annotation Term="Redfish.OwningEntity" String="DMTF"/>
+
+      <EntityType Name="CupsPolicy" BaseType="Resource.v1_0_0.Resource" Abstract="true">
+        <Annotation Term="OData.Description" String="The Policy schema contains properties related to a given CUPS Policy."/>
+        <Annotation Term="OData.LongDescription" String="The Policy schema contains properties related to a given CUPS Policy."/>
+        <Annotation Term="Capabilities.InsertRestrictions">
+          <Record>
+            <PropertyValue Property="Insertable" Bool="true"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.UpdateRestrictions">
+          <Record>
+            <PropertyValue Property="Updatable" Bool="true"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.DeleteRestrictions">
+          <Record>
+            <PropertyValue Property="Deletable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Redfish.Uris">
+          <Collection>
+            <String>/redfish/v1/Oem/Intel/CupsService/Policies/{PolicyId}</String>
+          </Collection>
+        </Annotation>
+      </EntityType>
+    </Schema>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsPolicy.v1_0_0">
+      <EntityType Name="CupsPolicy" BaseType="CupsPolicy.CupsPolicy">
+        <Property Name="Status" Type="Resource.Status" Nullable="false">
+          <Annotation Term="OData.Description" String="The status and health of the Resource and its subordinate or dependent Resources."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain any status or health properties of the Resource."/>
+        </Property>
+        <Property Name="AlertsEnabled" Type="Edm.Boolean" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Status of alert generation by this policy."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain value defining action to be taken when threshold is crossed. When set to true, alert will be sent to defined target."/>
+        </Property>
+        <Property Name="Threshold" Type="Edm.Decimal" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Threshold value for this policy."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain value of threshold for this policy."/>
+        </Property>
+        <Property Name="AveragingWindow" Type="Edm.Decimal" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Averaging window over which policy monitors its metrics."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain value averaging window over which policy monitors its metrics."/>
+          <Annotation Term="Validation.Minimum" Int="1"/>
+          <Annotation Term="Validation.Maximum" Int="10"/>
+          <Annotation Term="Measures.Unit" String="s"/>
+        </Property>
+        <Property Name="PolicyStorage" Type="CupsPolicy.v1_0_0.PolicyStorage" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Storage configuration of policy object."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain value defining configuration for policy storage option."/>
+        </Property>
+      </EntityType>
+
+      <EnumType Name="PolicyStorage">
+        <Member Name="Volatile">
+          <Annotation Term="OData.Description" String="Policy is only stored in volatile memory."/>
+        </Member>
+        <Member Name="Persistent">
+          <Annotation Term="OData.Description" String="Policy is stored in non-volatile memory."/>
+        </Member>
+      </EnumType>
+    </Schema>
+
+  </edmx:DataServices>
+</edmx:Edmx>
diff --git a/static/redfish/v1/schema/CupsSensorCollection_v1.xml b/static/redfish/v1/schema/CupsSensorCollection_v1.xml
new file mode 100644
index 00000000..e1df12c7
--- /dev/null
+++ b/static/redfish/v1/schema/CupsSensorCollection_v1.xml
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!---->
+<!--################################################################################       -->
+<!--# Redfish Schema:  CupsSensorCollection                                                -->
+<!--#                                                                                      -->
+<!--################################################################################       -->
+<!---->
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="OData"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Capabilities.V1.xml">
+    <edmx:Include Namespace="Org.OData.Capabilities.V1" Alias="Capabilities"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Resource_v1.xml">
+    <edmx:Include Namespace="Resource.v1_0_0"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/RedfishExtensions_v1.xml">
+    <edmx:Include Namespace="RedfishExtensions.v1_0_0" Alias="Redfish"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Sensor_v1.xml">
+    <edmx:Include Namespace="Sensor"/>
+  </edmx:Reference>
+
+  <edmx:DataServices>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsSensorCollection">
+      <Annotation Term="Redfish.OwningEntity" String="DMTF"/>
+
+      <EntityType Name="CupsSensorCollection" BaseType="Resource.v1_0_0.ResourceCollection">
+        <Annotation Term="OData.Description" String="A Collection of CUPS Sensor Resource instances."/>
+        <Annotation Term="OData.LongDescription" String="This Resource shall represent a Resource Collection of CUPS Sensor instances for a Redfish implementation."/>
+        <Annotation Term="Capabilities.InsertRestrictions">
+          <Record>
+            <PropertyValue Property="Insertable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.UpdateRestrictions">
+          <Record>
+            <PropertyValue Property="Updatable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.DeleteRestrictions">
+          <Record>
+            <PropertyValue Property="Deletable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Redfish.Uris">
+          <Collection>
+            <String>/redfish/v1/Oem/Intel/CupsService/Sensors</String>
+          </Collection>
+        </Annotation>
+        <NavigationProperty Name="Members" Type="Collection(Sensor.Sensor)">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/Read"/>
+          <Annotation Term="OData.Description" String="The members of this collection."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain an array of links to the members of this collection."/>
+          <Annotation Term="OData.AutoExpandReferences"/>
+          <Annotation Term="Redfish.Required"/>
+        </NavigationProperty>
+      </EntityType>
+
+    </Schema>
+  </edmx:DataServices>
+</edmx:Edmx>
diff --git a/static/redfish/v1/schema/CupsService_v1.xml b/static/redfish/v1/schema/CupsService_v1.xml
new file mode 100644
index 00000000..69b63134
--- /dev/null
+++ b/static/redfish/v1/schema/CupsService_v1.xml
@@ -0,0 +1,143 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!---->
+<!--################################################################################       -->
+<!--# Redfish Schema:  CupsService                                                         -->
+<!--#                                                                                      -->
+<!--################################################################################       -->
+<!---->
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="OData"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Capabilities.V1.xml">
+    <edmx:Include Namespace="Org.OData.Capabilities.V1" Alias="Capabilities"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Measures.V1.xml">
+    <edmx:Include Namespace="Org.OData.Measures.V1" Alias="Measures"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Resource_v1.xml">
+    <edmx:Include Namespace="Resource"/>
+    <edmx:Include Namespace="Resource.v1_0_0"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/RedfishExtensions_v1.xml">
+    <edmx:Include Namespace="RedfishExtensions.v1_0_0" Alias="Redfish"/>
+    <edmx:Include Namespace="Validation.v1_0_0" Alias="Validation"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="/redfish/v1/schema/CupsSensorCollection_v1.xml">
+    <edmx:Include Namespace="CupsSensorCollection"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="/redfish/v1/schema/CupsPolicyCollection_v1.xml">
+    <edmx:Include Namespace="CupsPolicyCollection"/>
+  </edmx:Reference>
+
+  <edmx:DataServices>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsService">
+      <Annotation Term="Redfish.OwningEntity" String="DMTF"/>
+
+      <EntityType Name="CupsService" BaseType="Resource.v1_0_0.Resource" Abstract="true">
+        <Annotation Term="OData.Description" String="The CupsService schema describes the root of the Cups Service. All other CUPS resources are linked directly or indirectly to the CUPS Service."/>
+        <Annotation Term="OData.LongDescription" String="The CupsService schema describes the root of the Cups Service. All other CUPS resources are linked directly or indirectly to the CUPS Service."/>
+        <Annotation Term="Capabilities.InsertRestrictions">
+          <Record>
+            <PropertyValue Property="Insertable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.UpdateRestrictions">
+          <Record>
+            <PropertyValue Property="Updatable" Bool="true"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Capabilities.DeleteRestrictions">
+          <Record>
+            <PropertyValue Property="Deletable" Bool="false"/>
+          </Record>
+        </Annotation>
+        <Annotation Term="Redfish.Uris">
+          <Collection>
+            <String>/redfish/v1/Oem/Intel/CupsService</String>
+          </Collection>
+        </Annotation>
+      </EntityType>
+    </Schema>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="CupsService.v1_0_0">
+      <Annotation Term="Redfish.OwningEntity" String="DMTF"/>
+      <Annotation Term="Redfish.Release" String="1.0"/>
+
+      <EntityType Name="CupsService" BaseType="CupsService.CupsService">
+        <Property Name="Status" Type="Resource.Status" Nullable="false">
+          <Annotation Term="OData.Description" String="The status and health of the Resource and its subordinate or dependent Resources."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain any status or health properties of the Resource."/>
+        </Property>
+        <Property Name="Interval" Type="Edm.Duration" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="The interval between CUPS readings."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain interval between CUPS readings."/>
+        </Property>
+        <NavigationProperty Name="CupsSensors" Type="CupsSensorCollection.CupsSensorCollection" ContainsTarget="true">
+          <Annotation Term="OData.Description" String="The link to the collection of CUPS sensors."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain a link to the collection of CUPS sensors."/>
+          <Annotation Term="OData.AutoExpandReferences"/>
+        </NavigationProperty>
+        <NavigationProperty Name="CupsPolicies" Type="CupsPolicyCollection.CupsPolicyCollection" ContainsTarget="true">
+          <Annotation Term="OData.Description" String="The link to the collection of metric report definitions."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain a link to the collection of metric report definitions."/>
+          <Annotation Term="OData.AutoExpandReferences"/>
+        </NavigationProperty>
+        <Property Name="AveragingPeriod" Type="Edm.Duration" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Averaging period over which CUPS Service shall average redings."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain averaging period over which CUPS Service shall average redings."/>
+        </Property>
+        <Property Name="LoadFactorConfiguration" Type="CupsService.v1_0_0.LoadFactorConfiguration" Nullable="false">
+          <Annotation Term="OData.Description" String="Describes which load factors should be used for CUPS Index calculations."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain description of which load factors should be used for CUPS Index calculations."/>
+        </Property>
+        <Property Name="DynamicLoadFactors" Type="CupsService.v1_0_0.LoadFactors" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/Read"/>
+          <Annotation Term="OData.Description" String="Contains the dynamic load factors."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain the dynamic load factors."/>
+        </Property>
+        <Property Name="StaticLoadFactors" Type="CupsService.v1_0_0.LoadFactors" Nullable="false">
+          <Annotation Term="OData.Permissions" EnumMember="OData.Permission/ReadWrite"/>
+          <Annotation Term="OData.Description" String="Contains the dynamic load factors."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain the dynamic load factors."/>
+        </Property>
+      </EntityType>
+
+      <EnumType Name="LoadFactorConfiguration">
+        <Annotation Term="OData.Description" String="Selects which load factors shall be used by CUPS Service."/>
+        <Annotation Term="OData.LongDescription" String="This property shall determine whether the CUPS Service shall use dynamic (calculated in runtime) or static (provided by the user e.g. datacenter administrator) load factors."/>
+
+        <Member Name="Dynamic">
+          <Annotation Term="OData.Description" String="CUPS service shall use load factors calculated in runtime."/>
+        </Member>
+        <Member Name="Static">
+          <Annotation Term="OData.Description" String="CUPS service shall use load factors provided by the user."/>
+        </Member>
+      </EnumType>
+
+      <ComplexType Name="LoadFactors">
+        <Annotation Term="OData.Description" String="Stores values of load factors."/>
+        <Annotation Term="OData.LongDescription" String="This type shall contain values of load factors used by the CUPS Service."/>
+        <Annotation Term="OData.AdditionalProperties" Bool="false"/>
+
+         <Property Name="CoreLoadFactor" Type="Edm.Decimal" Nullable="true">
+          <Annotation Term="OData.Description" String="Load factor for Core compute domain."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain load factor for Core compute domain."/>
+        </Property>
+        <Property Name="MemoryLoadFactor" Type="Edm.Decimal" Nullable="true">
+          <Annotation Term="OData.Description" String="Load factor for Memory compute domain."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain load factor for Memory compute domain."/>
+        </Property>
+        <Property Name="IioLoadFactor" Type="Edm.Decimal" Nullable="true">
+          <Annotation Term="OData.Description" String="Load factor for IIO compute domain."/>
+          <Annotation Term="OData.LongDescription" String="This property shall contain load factor for IIO compute domain."/>
+        </Property>
+      </ComplexType>
+    </Schema>
+
+  </edmx:DataServices>
+</edmx:Edmx>
diff --git a/static/redfish/v1/schema/OemServiceRoot_v1.xml b/static/redfish/v1/schema/OemServiceRoot_v1.xml
new file mode 100644
index 00000000..490efe2f
--- /dev/null
+++ b/static/redfish/v1/schema/OemServiceRoot_v1.xml
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<edmx:Edmx xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx" Version="4.0">
+
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Core.V1.xml">
+    <edmx:Include Namespace="Org.OData.Core.V1" Alias="OData"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://docs.oasis-open.org/odata/odata/v4.0/errata03/csd01/complete/vocabularies/Org.OData.Measures.V1.xml">
+    <edmx:Include Namespace="Org.OData.Measures.V1" Alias="Measures"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/RedfishExtensions_v1.xml">
+    <edmx:Include Namespace="RedfishExtensions.v1_0_0" Alias="Redfish"/>
+  </edmx:Reference>
+  <edmx:Reference Uri="http://redfish.dmtf.org/schemas/v1/Resource_v1.xml">
+    <edmx:Include Namespace="Resource"/>
+    <edmx:Include Namespace="Resource.v1_0_0"/>
+  </edmx:Reference>
+
+  <edmx:Reference Uri="/redfish/v1/schema/CupsService_v1.xml">
+    <edmx:Include Namespace="CupsService"/>
+  </edmx:Reference>
+
+  <edmx:DataServices>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="OemServiceRoot">
+      <Annotation Term="Redfish.OwningEntity" String="OpenBMC"/>
+    </Schema>
+
+    <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="OemServiceRoot.v1_0_0">
+      <Annotation Term="Redfish.OwningEntity" String="OpenBMC"/>
+      <Annotation Term="Redfish.Release" String="1.0"/>
+
+        <ComplexType Name="Oem" BaseType="Resource.OemObject">
+          <Annotation Term="OData.AdditionalProperties" Bool="true"/>
+          <Annotation Term="OData.Description" String="OemServiceRoot Oem properties."/>
+          <Annotation Term="OData.AutoExpand"/>
+          <Property Name="Intel" Type="OemServiceRoot.v1_0_0.Intel"/>
+        </ComplexType>
+
+        <ComplexType Name="Intel" BaseType="Resource.OemObject">
+          <Annotation Term="OData.AdditionalProperties" Bool="true"/>
+          <Annotation Term="OData.Description" String="Oem properties for Intel."/>
+          <Annotation Term="OData.AutoExpand"/>
+          <NavigationProperty Name="CupsService" Type="CupsService.CupsService" Nullable="false">
+            <Annotation Term="OData.Permissions" EnumMember="OData.Permission/Read"/>
+            <Annotation Term="OData.Description" String="The link to Intel Memory Resilience Technology application."/>
+            <Annotation Term="OData.LongDescription" String="This property shall contain the link to Intel Memory Resilience Technology application."/>
+            <Annotation Term="OData.AutoExpandReferences"/>
+          </NavigationProperty>
+        </ComplexType>
+    </Schema>
+
+  </edmx:DataServices>
+</edmx:Edmx>
-- 
2.25.1

